src/main/kotlin/no/fdk/dataservicecatalog/domain/DataService.kt

```diff
@@ -1,27 +1,46 @@
 package no.fdk.dataservicecatalog.domain
 
-import com.fasterxml.jackson.annotation.JsonProperty
 import jakarta.validation.Valid
 import jakarta.validation.constraints.NotBlank
 import jakarta.validation.constraints.NotEmpty
+import org.springframework.data.annotation.CreatedDate
+import org.springframework.data.annotation.Id
+import org.springframework.data.annotation.LastModifiedDate
+import org.springframework.data.annotation.Version
+import org.springframework.data.mongodb.core.mapping.Document
+import java.time.LocalDateTime
 
 /*
 https://data.norge.no/specification/dcat-ap-no#Datatjeneste
  */
 
+@Document(collection = "dataServices")
 data class DataService(
 
     /*
-    endepunktsURL (dcat:endpointURL)
+    identifikator (dct:identifier)
      */
-    @field:NotBlank(message = "Cannot be null or blank")
-    val endpointUrl: String? = null,
+    @Id
+    val id: String? = null,
+
+    @CreatedDate
+    val created: LocalDateTime? = null,
+
+    @LastModifiedDate
+    val modified: LocalDateTime? = null,
+
+    @Version
+    val version: Int? = null,
+
+    val catalogId: String? = null,
+
+    val status: Status? = null,
 
     /*
-    identifikator (dct:identifier)
+    endepunktsURL (dcat:endpointURL)
      */
-    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
-    val identifier: String? = null,
+    @field:NotBlank(message = "Cannot be null or blank")
+    val endpointUrl: String? = null,
```

Comment:
Kan vel ha dette som ikke nullbar når du uansett har kravet "NotBlank"?
```suggestion
    val endpointUrl: String,
```

Reply:
Same her jf. validering.